### ISSUE TITLE
Clippy matches redundant closure to Option creation

### ISSUE THREAD

**Issue by garentyler** [rust-lang/rust-clippy#9129]
### Summary

Clippy thinks that the option creation syntax `Some(x)` is a function when it shouldn't.

### Reproducer

I tried this code (context is irrelevant):

```rust
Self::from_row(&rows[0]).map(|x| Some(x))
```

I saw this happen:

```
warning: redundant closure
   --> src/db/user.rs:185:38
    |
185 |         Self::from_row(&rows[0]).map(|x| Some(x))
    |                                      ^^^^^^^^^^^ help: replace the closure with the function itself: `Some`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure
```

I expected to see this happen:
No output.

### Version

```text
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: aarch64-apple-darwin
release: 1.62.0
LLVM version: 14.0.5
```

**Comment by Jarcho** [comment]
All tuple constructors are functions. `.map(Some)` works just fine.

**Comment by y21** [comment]
Closing as it doesn't look like there's an actual FP here.
